pkg/collector/container.go

```diff
@@ -76,11 +76,6 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 
 	if len(otelcol.Spec.VolumeMounts) > 0 {
 		volumeMounts = append(volumeMounts, otelcol.Spec.VolumeMounts...)
-	} else if otelcol.Spec.Mode == "statefulset" {
```

Comment:
so the users will have to define the volumes explicitly in the CR? 

I don't think we have even documented this default volume so not sure somebody used it.

Reply:
If you need a volume you're need to declare one to have one yes. Basically if not you Will always end up having PVCs / vols required that in some circumstances is not needed by default